### ISSUE TITLE
MODEL_NAME, ARCH vs different backbones

### ISSUE THREAD

**Issue by chg-87** [facebookresearch/SlowFast#683]
Hi guys!
First of all, thank you for this amazing repo!
I have questions regarding terminology, cause I got slightly confused by the names `MODEL.ARCH` and `MODEL.MODEL_NAME` and this part of the code:
```
# Supported model types
_MODEL_TYPES = {
    "slowfast": SlowFast,
    "slow": ResNet,
    "c2d": ResNet,
    "i3d": ResNet,
    "slow_c2d": ResNet,
    "x3d": X3D,
    "mvit": MViT,
}
```
Based on the description I thought that I could choose the backbone from any of the mentioned above in SlowFast architecture and then e.g. I can have SlowFast architecture with an X3D backbone or I have SlowFast architecture with MViT backbone to extract features?

But what if I choose SlowFast as a backbone? Then I have SlowFast architecture with SlowFast backbone, but what is the backbone of my backbone then - is it ResNet as described in the article or does it work in a different way? I don't really get this part... Or I just choose MViT and it's in no way connected to SlowFast and if I want I can compare it to SlowFast, but why the backbone then?

Second of all, what is the classifier at the end? Is it e.g. a sigmoid-based classifier layer, as it was written in the paper? Is it specified by HEAD_ACT?

Lastly, I wanted to ask why the `MODEL_NAME` for `ARCH`s I3D, C2D is ResNet, and for SlowFast is SlowFast, but maybe the answer to the first question clarifies it all.

I really appreciate any help!

**Comment by alpargun** [comment]
Architectures:

- x3d
- slowfast
- slow
- mvit
- c2d
- i3d

X3D is based from ResNet but a more efficient network, and hence, you can use them interchangeably (with an efficiency vs. accuracy trade-off). Hence, you can use X3D or ResNet for the architectures: 
- x3d
- slowfast
- slow
- c2d
- i3d

Note that these are all architectures and X3D and ResNet are the backbone networks. Architectures determine the number of pathways (e.g. slow and fast pathways, non-local blocks etc.), which affect the training scheme. For more details, you can refer to the corresponding papers.

Furthermore, there is also the "TASK" setting, which you can configure as  "SSL" if you would like a self-supervised training setup.

As far as I know, MViT backbone cannot be used with the above architectures and it should be set as:

```
MODEL:
  ARCH: mvit
  MODEL_NAME: MViT
```

For more config examples, you can look at the [configs/ directory](https://github.com/alpargun/SlowFast/tree/main/configs).
